### PR TITLE
[pysrc2cpg] Added auto-discovery for venv directories

### PR DESCRIPTION
**Pull request by max-leuthaeuser**
- directories with a `pyvenv.cfg` file are ignored if `ignoreVenvDir == true`
- nothing is ignored if `ignoreVenvDir == false`
- if `ignoreVenvDir == true` and there is at least one folder set via old `--venvDir` or new `--venvDirs`: only these are ignored; no auto-discovery then

### CODE REVIEW

File: joern-cli/frontends/pysrc2cpg/src/main/scala/io/joern/pysrc2cpg/Py2CpgOnFileSystem.scala [hunk 1]
```diff
@@ -93,6 +97,22 @@ class Py2CpgOnFileSystem extends X2CpgFrontend[Py2CpgOnFileSystemConfig] {
     }
   }
 
+  private def filterEnvDir(config: Py2CpgOnFileSystemConfig, file: Path, inputPath: Path): Boolean = {
```
Comment (maltek) [inline joern-cli/frontends/pysrc2cpg/src/main/scala/io/joern/pysrc2cpg/Py2CpgOnFileSystem.scala:100]: just nit-picking, but I find this function a bit hard to follow: I had to look at the place where this is called to figure out what the return values actually mean. I think it might be clearer if we flip the returned bool around and rename the function to `isAutoDectedVenv`? (And adjust the call-site to use `filterNot`.)

File: joern-cli/frontends/pysrc2cpg/src/test/scala/io/joern/pysrc2cpg/config/VenvExcludeTests.scala [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+        Set(
+          ".venv/a1.py",
+          ".venv/sub/s1.py",
+          ".venv2/a2.py",
+          ".venv2/sub/s2.py",
+          "folder/b.py",
+          "folder/c.py",
+          "main.py"
+        )
+      ),
```
Comment (maltek) [inline joern-cli/frontends/pysrc2cpg/src/test/scala/io/joern/pysrc2cpg/config/VenvExcludeTests.scala:75]: doesn't really matter, but we could write these tests as

```scala
"exclude nothing if ignoreVenvDir is false" should {
   assert(testWithArguments(false, Seq.empty, Seq(...)))
  }
}
```

should be a little shorter in the end, and result in somewhat nicer errors when the tests fail

**Review by maltek — APPROVED** [review]

File: joern-cli/frontends/pysrc2cpg/src/main/scala/io/joern/pysrc2cpg/Py2CpgOnFileSystem.scala (hunk 1 above)
Comment (max-leuthaeuser) [reply joern-cli/frontends/pysrc2cpg/src/main/scala/io/joern/pysrc2cpg/Py2CpgOnFileSystem.scala:100]: I wanted to keep it consistent to `filterIgnoreDirNames` down below (also their callsites).
But you are right, I guess both could benefit from a small refactoring there for readibility.

File: joern-cli/frontends/pysrc2cpg/src/test/scala/io/joern/pysrc2cpg/config/VenvExcludeTests.scala (hunk 2 above)
Comment (maltek) [reply joern-cli/frontends/pysrc2cpg/src/test/scala/io/joern/pysrc2cpg/config/VenvExcludeTests.scala:75]: OK, leave it as-is then.

I've had some problems figuring out which table entry was failing for some scalatest tests in the past, so my knee-jerk reaction is to avoid them. But this might very well not apply here.

File: joern-cli/frontends/pysrc2cpg/src/main/scala/io/joern/pysrc2cpg/Py2CpgOnFileSystem.scala (hunk 1 above)
Comment (max-leuthaeuser) [reply joern-cli/frontends/pysrc2cpg/src/main/scala/io/joern/pysrc2cpg/Py2CpgOnFileSystem.scala]: I implemented the refactoring you suggested.

**Review by maltek — APPROVED** [review]
